go/ros/ros2db3_to_mcap.go

```diff
@@ -72,6 +72,13 @@ func getSchemas(encoding string, directories []string, types []string) (map[stri
 		for len(subdefinitions) > 0 {
 			subdefinition := subdefinitions[0]
 			if !first {
+				// if the previous write did not end with a newline, add one now
+				if messageDefinition.Bytes()[messageDefinition.Len()-1] != '\n' {
+					_, err := messageDefinition.WriteString("\n")
```

Comment:
nit: could use WriteByte since you're reading a byte on the prev line, doesn't really matter though